### PR TITLE
Backport of #4982 (Fix storing and retrieving the running means and variances of the BacthNorm layer)

### PR DESCRIPTION
**Pull request by lmoneta**

### CODE REVIEW

**Comment by lmoneta** [comment]
@ashlaban or @stwunsch can you please review this ?

File: tmva/tmva/inc/TMVA/DNN/BatchNormLayer.h [hunk 1]
```diff
@@ -304,6 +317,12 @@ auto TBatchNormLayer<Architecture_t>::Forward(Tensor_t &x, bool inTraining) -> v
    }
 
    else {
+      // if (fTrainedBatches > 0) {
```
Comment (stwunsch) [inline tmva/tmva/inc/TMVA/DNN/BatchNormLayer.h]: You have still outcommented debug code in here!

File: tmva/tmva/src/MethodDL.cxx [hunk 2]
```diff
@@ -1551,14 +1545,11 @@ void MethodDL::TrainDeepNet()
...
-                  const auto & nLayer = fNet->GetLayerAt(i);
-                  const auto & dLayer = deepNet.GetLayerAt(i);
-                  ArchitectureImpl_t::CopyDiffArch(nLayer->GetWeights(), dLayer->GetWeights() );
-                  ArchitectureImpl_t::CopyDiffArch(nLayer->GetBiases(), dLayer->GetBiases() );
-                  // std::cout << "Weights for layer " << i << std::endl;
-                  // for (size_t k = 0; k < dlayer->GetWeights().size(); ++k)
-                  //    dLayer->GetWeightsAt(k).Print();
-                  // debug tensors
+                  fNet->GetLayerAt(i)->CopyParameters(*deepNet.GetLayerAt(i));
+                  // if (i == 0 && deepNet.GetLayerAt(0)->GetWeights().size() > 1) {
```
Comment (stwunsch) [inline tmva/tmva/src/MethodDL.cxx]: Same here.

File: tmva/tmva/src/MethodDL.cxx [hunk 3]
```diff
@@ -1855,10 +1846,11 @@ std::vector<Double_t> MethodDL::PredictDeepNet(Long64_t firstEvt, Long64_t lastE
 
    // copy weights from the saved fNet to the built DeepNet
    for (size_t i = 0; i < deepNet.GetDepth(); ++i) {
-      const auto & nLayer = fNet->GetLayerAt(i);
-      const auto & dLayer = deepNet.GetLayerAt(i);
-      Architecture_t::CopyDiffArch(dLayer->GetWeights(), nLayer->GetWeights() );
-      Architecture_t::CopyDiffArch(dLayer->GetBiases(), nLayer->GetBiases() );
+      deepNet.GetLayerAt(i)->CopyParameters(*fNet->GetLayerAt(i));
+      // if (i == 0 && deepNet.GetLayerAt(0)->GetWeights().size() > 1) {
```
Comment (stwunsch) [inline tmva/tmva/src/MethodDL.cxx]: And here.

File: tmva/tmva/inc/TMVA/DNN/Adam.h [hunk 4]
```diff
@@ -70,7 +70,7 @@ class TAdam : public VOptimizer<Architecture_t, Layer_t, DeepNet_t> {
 public:
    /*! Constructor. */
    TAdam(DeepNet_t &deepNet, Scalar_t learningRate = 0.001, Scalar_t beta1 = 0.9, Scalar_t beta2 = 0.999,
-         Scalar_t epsilon = 1e-8);
+         Scalar_t epsilon = 1e-7);
```
Comment (stwunsch) [inline tmva/tmva/inc/TMVA/DNN/Adam.h]: You chang ehere a default for Adam, this was intended?

Comment (lmoneta) [reply tmva/tmva/inc/TMVA/DNN/Adam.h]: I have put same default as used in Tensorflow, 
see https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adam
Since we are using the same convention. Epsilon is actually epsilon_hat defined in formula before paragraph 2.1 in Adam paper, https://arxiv.org/pdf/1412.6980.pdf

Comment (lmoneta) [reply tmva/tmva/inc/TMVA/DNN/Adam.h]: Thank you for the review. 
I will remove the commented line in one of the mext commits
